app/database/operator/server_data_operator/handlers/index.ts

```diff
@@ -53,11 +54,20 @@ export default class ServerDataOperatorBase extends BaseDataOperator {
             return [];
         }
 
+        const names = Array.from(new Set(emojis.map((e) => e.name)));
+
+        const locals = await queryCustomEmojisByName(this.database, names).fetch();
+
+        const deleteRawValues = locals.
+            filter((local) => !emojis.find((raw) => raw.id === local.id)).
```

Comment:
This is not efficient, why don't you keep using the set instead of transforming back to an array?

Reply:
Changed the code slightly to make it more performant, can you review it again?

I can't use the original Set, because the idea here is:
- Search WatermelonDB by the emoji names to see if there is any with the same name that comes from Client/Server
- If there is nothing on WatermelonDB, nothing will be done
- If there is emoji with the same name on WatermelonDB, I want to see if the actual ID is different, if the id is different, we need to remove the old one, which means probably the client/server emoji was recreated with a different image.

I changed the code to create another set with the ids, so we don't need to make n^2 filtering in this case.